File: src/main/java/org/tdf/lotusvm/types/MemorySection.java
package org.tdf.lotusvm.types;

import lombok.Getter;
import org.tdf.lotusvm.common.BytesReader;

import java.util.List;

/**
 * The memory section has the id 5. It decodes into a vector of memories that represent the mems component of a module.
 */
public class MemorySection extends AbstractSection {

    @Getter
    private List<LimitType> memories;

    public MemorySection(SectionID id, long size, BytesReader payload, int offset, int limit) {
        super(id, size, payload, offset, limit);
    }

    @Override
    void readPayload() {
        memories = LimitType.readLimitTypesFrom(getReader());
    }
}

File: src/main/java/org/tdf/lotusvm/types/MemorySection.kt
package org.tdf.lotusvm.types

import org.tdf.lotusvm.common.BytesReader

/**
 * The memory section has the id 5. It decodes into a vector of memories that represent the mems component of a module.
 */
class MemorySection(id: SectionID, size: Long, payload: BytesReader, offset: Int, limit: Int) :
    AbstractSection(id, size, payload, offset, limit) {
    var memories: List<LimitType> = emptyList()
        private set

    public override fun readPayload() {
        memories = reader.readObjectVec(LimitType.Companion)
    }
}